File: 1004-最大连续1的个数III.cpp
#include <iostream>
using namespace std;
#include <vector>
#include <algorithm>
class Solution
{
public:
    int longestOnes(vector<int> &nums, int k)
    {
        int res = 0, left = 0, sum = 0;
        for (int right = 0; right < nums.size(); right++)
        {
            sum += nums[right] != 1;
            while (sum > k)
            {
                sum -= nums[left] != 1;
                left++;
            }
            res = max(res, right - left + 1);
        }
        return res;
    }
};

File: 1004-最大连续1的个数III.py
from typing import List


class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        res, left, sum = 0, 0, 0
        for right in range(len(nums)):
            sum += nums[right] != 1
            while sum > k:
                sum -= nums[left] != 1
                left += 1
            res = max(res, right - left + 1)
        return res
